### PR TITLE
feat/create_update_delete_booking

### PR DESCRIPTION
**Pull request by HaiYenNguyen12**
feat: create booking(calculate amount, add UserId)
        update booking
        delete booking

### CODE REVIEW

File: HotelBooking.Service/Services/BookingService.cs [hunk 1]
```diff
@@ -13,29 +16,33 @@ public class BookingService : IBookingService
...
             this.roomRepository = roomRepository;
             this.mapper = mapper;
             this.unitOfWork = unitOfWork;
+            this.roomServiceRepository = roomServiceRepository;
         }
 
-        public async Task<bool> AddBookingAsync(BookingVM model)
+        public async Task<bool> AddBookingAsync(BookingVM model,string UserId)
         {
             var booki = new Booking()
```
Comment (orienthoanguyen) [inline HotelBooking.Service/Services/BookingService.cs:39]: what does booki mean?

Comment (HaiYenNguyen12) [reply HotelBooking.Service/Services/BookingService.cs]: Sorry, it's my mistake. I will fix it now.

File: HotelBooking.API/Controllers/BookingController.cs [hunk 2]
```diff
@@ -41,15 +45,30 @@ public async Task<IActionResult> CreateBookingAsync([FromForm] BookingVM model)
                 logger.LogError("Invalid model sent from client.");
                 return BadRequest("Invalid model object");
             }
-            var res = await bookingService.AddBookingAsync(model);
+            var id = currentUser.UserId;
```
Comment (orienthoanguyen) [inline HotelBooking.API/Controllers/BookingController.cs:48]: Do we need to declare id var in this case?

File: HotelBooking.Service/Services/BookingService.cs [hunk 3]
```diff
@@ -13,41 +16,45 @@ public class BookingService : IBookingService
...
             this.bookedRoomRepository = bookedRoomRepository;
             this.bookingRepository = bookingRepository;
             this.roomRepository = roomRepository;
             this.mapper = mapper;
             this.unitOfWork = unitOfWork;
+            this.roomServiceRepository = roomServiceRepository;
         }
 
-        public async Task<bool> AddBookingAsync(BookingVM model)
+        public async Task<bool> AddBookingAsync(BookingVM model,string UserId)
```
Comment (orienthoanguyen) [inline HotelBooking.Service/Services/BookingService.cs:37]:
```suggestion
        public async Task<bool> AddBookingAsync(BookingVM model,string userId)
```

File: HotelBooking.Service/Services/BookingService.cs [hunk 4]
```diff
@@ -56,11 +63,61 @@ public async Task<bool> CheckValidationDurationForRoom(DurationVM model, string
             return await bookedRoomRepository.CheckValidation(model, new Guid(roomId));
         }
 
+        public async Task<bool> DeleteBookingAsync(string Id)
```
Comment (orienthoanguyen) [inline HotelBooking.Service/Services/BookingService.cs:66]: Id is Guid then force the param type to Guid please

File: HotelBooking.Service/Services/BookingService.cs [hunk 5]
```diff
@@ -56,11 +63,61 @@ public async Task<bool> CheckValidationDurationForRoom(DurationVM model, string
...
+        {
+            var res = await bookingRepository.GetByIdAsync(new Guid(Id));
+            if (res == null) return false;
+            res.DeletedDate = DateTime.Now;
+            res.IsDeleted= true;
+            await unitOfWork.SaveAsync();
+            return true;
```
Comment (orienthoanguyen) [inline HotelBooking.Service/Services/BookingService.cs:73]:
```suggestion
            var res = await bookingRepository.GetByIdAsync(new Guid(Id));
            
            if (res == null) return false;
            
            res.DeletedDate = DateTime.Now;
            res.IsDeleted= true;
            await unitOfWork.SaveAsync();
            
            return true;
```

File: HotelBooking.API/Controllers/BookingController.cs (hunk 2 above)
Comment (HaiYenNguyen12) [reply HotelBooking.API/Controllers/BookingController.cs:48]: Do you mean the user can create a booking when they don't have an account?
I think they must register an account before they book a hotel.

**Review by orienthoanguyen — APPROVED** [review]

**Pull request merged**